codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -288,7 +288,12 @@ private int maxUsableChannelBytes() {
         return min(connectionState().windowSize(), useableBytes);
     }
 
-    private int writableBytes(int requestedBytes) {
+    /**
+     * Package private for testing purposes only!
+     * @param requestedBytes The desired amount of bytes.
+     * @return The amount of bytes that can be supported by underlying {@link Channel} without queuing "too-much".
+     */
+    final int writableBytes(int requestedBytes) {
```

Comment:
I added some writabilty change stuff at the last minute that actually wrecked the test. Check updated test.

The key is that we need the streamableBytes to be greater than the amount will will attempt to send. The amount we will be allowed to send is determined by this method.